code/datums/ammo/bullet/tank.dm

```diff
@@ -72,3 +77,32 @@
 	for(var/mob/living/carbon/L in T)
 		if(L.stat == CONSCIOUS && L.mob_size <= MOB_SIZE_XENO)
 			shake_camera(L, 1, 1)
+
+//Minigun Ammo//
+
+/datum/ammo/bullet/tank/minigun
+	name = "minigun bullet"
+	headshot_state = HEADSHOT_OVERLAY_MEDIUM
+
+	accuracy = -HIT_ACCURACY_TIER_1
+	accuracy_var_low = PROJECTILE_VARIANCE_TIER_8
+	accuracy_var_high = PROJECTILE_VARIANCE_TIER_8
+	accurate_range = 12
+	damage = 35
+	penetration = ARMOR_PENETRATION_TIER_6
+
+/datum/ammo/bullet/tank/minigun/New()
+	..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
+		damage = 15
+	else if(SSticker.current_state < GAME_STATE_PLAYING)
+		RegisterSignal(SSdcs, COMSIG_GLOB_MODE_PRESETUP, PROC_REF(setup_hvh_damage))
+
+/datum/ammo/bullet/tank/minigun/proc/setup_hvh_damage()
```

Comment:
Every proc that is registered in register signal must have SIGNAL_HANDLER on the top.